themes/wporg-translate-events-2024/blocks/event-flag/index.php

```diff
@@ -0,0 +1,40 @@
+<?php
+namespace Wporg\TranslationEvents\Theme_2024;
+use Wporg\TranslationEvents\Translation_Events;
+
+register_block_type(
+	'wporg-translate-events-2024/event-flag',
+	array(
+		// phpcs:ignore Generic.CodeAnalysis.UnusedFunctionParameter.Found
+		'render_callback' => function ( array $attributes, $content, $block ) {
+			if ( ! isset( $block->context['postId'] ) ) {
+				return '';
+			}
+			$event_id = get_the_ID();
+			$event = Translation_Events::get_event_repository()->get_event( $event_id );
+			if ( ! $event ) {
+				return $event_id;
+			}
+
+			global $current_user_attendee_per_event;
```

Comment:
I think we shouldn't rely on global variables, as the block can thus not be called without knowing that this variable needs to be set. Not sure what the correct solution is though, maybe the `Attendee_Repository` also needs caching?